src/components/Sidebars/Payback.js

```diff
@@ -123,22 +123,46 @@ export default Payback;
 const checkForProxyAndAllowance = async (
   maker,
   updateState,
-  setHasAllowance
+  setHasAllowance,
+  madeInitialCheck
 ) => {
   const proxyAddress = await maker.service('proxy').getProxyAddress();
   const token = maker.getToken('MDAI');
-  const hasAllowance =
-    proxyAddress &&
+
+  const checkAllowance = async () =>
+    !!proxyAddress &&
     (await token.allowance(maker.currentAddress(), proxyAddress)).eq(
       MAX_UINT_BN
     );
 
+  const pollCheckAllowance = async () => {
+    let numTries = 5;
+    let allowance;
+    while (numTries > 0) {
+      numTries--;
+      allowance = await checkAllowance();
+      if (allowance) return allowance;
+    }
+    return false;
+  };
+
+  let hasAllowance = await checkAllowance();
+  if (madeInitialCheck && !hasAllowance) {
+    hasAllowance = await pollCheckAllowance();
+  }
+
   setHasAllowance(hasAllowance);
-  updateState({
-    proxyAddress,
-    startedWithoutProxy: !proxyAddress,
-    startedWithoutAllowance: !hasAllowance
-  });
+
+  if (madeInitialCheck) {
+    updateState({ proxyAddress });
+  } else {
+    updateState({
+      proxyAddress,
+      startedWithoutProxy: !proxyAddress,
+      startedWithoutAllowance: !hasAllowance,
+      madeInitialCheck: !madeInitialCheck
+    });
+  }
```

Comment:
The `madeInitialCheck` serves as a flag to say "is this the first time we have ran this function?". The reason for this is that the `startedWithoutProxy/Allowance` flags should only be set the first time we check for the existence of a proxy and allowance. Before, if this function was called a second time they would be overwritten and the toggles would be removed where they should have persisted.